pkg/rolling/restart_handler.go

```diff
@@ -0,0 +1,109 @@
+package rolling
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ydb-platform/ydb-go-genproto/draft/protos/Ydb_Maintenance"
+	"go.uber.org/zap"
+
+	"github.com/ydb-platform/ydbops/pkg/rolling/restarters"
+)
+
+type restartStatus struct {
+	nodeID uint32
+	as     *Ydb_Maintenance.ActionState
+	err    error
+}
+
+type restartHandler struct {
+	logger    *zap.SugaredLogger
+	queue     chan *Ydb_Maintenance.ActionGroupStates
+	restarter restarters.Restarter
+	statusCh  chan<- restartStatus
+
+	// TODO(shmel1k@): probably, not needed here.
+	nodes map[uint32]*Ydb_Maintenance.Node
+
+	done chan struct{}
+	wg   sync.WaitGroup
+
+	maxConcurrentRestarts int
+}
+
+func (rh *restartHandler) push(state *Ydb_Maintenance.ActionGroupStates) {
+	select {
```

Comment:
I am also not convinced this is necessary. Push should be non-blocking and just add the task to a bufferized queue. You are checking on `done` anyways in line 50, in function `run`, just before trying to take one more task.